### PR TITLE
Image classification training with RecordIter

### PR DESCRIPTION
**Pull request by hetong007**
This PR includes the following changes:

- A tutorial on training with record iter
- Improved script to prepare record files for imagenet
- New training lr scheduler
  - step, or piecewise-constant
  - poly
  - cosine
  - All above include warmup option
- Updated training scripts for imagenet
- Update resnet_v1b with the possibility to initialize the gamma to zero for the last BN layer in each bottleneck block.

### CODE REVIEW

**Comment by hetong007** [comment]
This PR is on hold, waiting for https://github.com/apache/incubator-mxnet/pull/11027 and https://github.com/apache/incubator-mxnet/pull/11327 . 

The blocking factors are newly trained resnet152_v1 and resnet152_v2. Will also update the yml files to incorporate the recent change on `save_parameters`.

**Comment by mli** [comment]
Job PR-161-37 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-161/37/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-161-38 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-161/38/index.html 
Code coverage of this PR:  vs. Master:

**Comment by hetong007** [comment]
It's in a good shape now. Reviews are very welcome @piiswrong @zhreshold @zhanghang1989

File: docs/model_zoo/index.rst [hunk 1]
```diff
@@ -55,30 +55,46 @@ Besides the listed, we provide more models trained on ImageNet in the upstream
 
 .. hint::
 
-    Training commands work with this script: :download:`Download train_imagenet.py<../../scripts/classification/imagenet/train_imagenet.py>`
+    Training commands work with these two script:
+
+    :download:`Download train_imagenet.py<../../scripts/classification/imagenet/train_imagenet.py>`
+
+    :download:`Download utils.py<../../scripts/classification/imagenet/utils.py>`
```
Comment (zhreshold) [inline docs/model_zoo/index.rst]: IMO, using two separate script files isn't a good idea.

File: docs/model_zoo/index.rst [hunk 2]
```diff
@@ -55,30 +55,46 @@ Besides the listed, we provide more models trained on ImageNet in the upstream
...
+
+    The `resnet_v1b` family is a modified version of `resnet_v1`, specifically we set stride at the 3x3 layer for a bottleneck block. `ResNet18` and `ResNet34` have identical `v1` and `v1b` network structures. This modification is widely mentioned in literatures [8]_ .
```
Comment (zhreshold) [inline docs/model_zoo/index.rst]: "widely mentioned" seems to be a bit weird.

**Comment by mli** [comment]
Job PR-161-39 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-161/39/index.html 
Code coverage of this PR:  vs. Master:

**Comment by mli** [comment]
Job PR-161-40 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-161/40/index.html 
Code coverage of this PR:  vs. Master:

**Comment by hetong007** [comment]
Going to merge it. Further reviews are welcome and I'll address them in another PR.

**Pull request merged**